### PR TITLE
Task6

### PR DESCRIPTION
**Pull request by GlebKast**

### CODE REVIEW

File: UI/scripts/scriptWithDOM.js [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+            this._photoPosts = photoPosts;
+            this.currentUser = JSON.parse(localStorage.getItem('current_user'));
+            this.numOfPosts = 0;
+            for (let i = 0; i < this._photoPosts.length; i++) {
+                if (this._validatePost(this._photoPosts[i])) {
+                    this.numOfPosts++;
+                    localStorage.setItem('posts_number', JSON.stringify(this.numOfPosts));
+                    this._photoPosts[i].id = this.newID(this.numOfPosts);
+                    document.getElementById('posts').innerHTML = this.createHTMLPost(this._photoPosts[i]) + document.getElementById('posts').innerHTML;
+                    localStorage.setItem(this._photoPosts[i].id, JSON.stringify(this._photoPosts[i]));
```
Comment (ala-n) [inline UI/scripts/scriptWithDOM.js]: Why don't you store whole model as one item ?

File: UI/scripts/scriptWithDOM.js [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+            }
+        } else{
+            this._photoPosts =[];
+            this.numOfPosts = JSON.parse(localStorage.getItem('posts_number'));
+            this.currentUser = JSON.parse(localStorage.getItem('current_user'));
+        }
+    }
+
+    getPostById(idP) {
+        return JSON.parse(localStorage.getItem(idP));
```
Comment (ala-n) [inline UI/scripts/scriptWithDOM.js]: uncached access, not according conditions of lab

File: UI/scripts/scriptWithDOM.js [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+            console.log("wrong post photoLink");
+            return false;
+        }
+        return true;
+    }
+
+    createHTMLPost(post) {
+        if (this.currentUser === post.author) {
+            return `
+            <div class="post" id="${post.id}">
```
Comment (ala-n) [inline UI/scripts/scriptWithDOM.js]: Copypaste of markup for authorized/unauthorized state, should be optimized

File: UI/scripts/scriptWithDOM.js [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+                </div>
+                <img src="${post.photoLink}" alt="photo">
+                <div class="post_buttons">
+                    <button class="like"></button>
+                    <button class="edit"></button>
+                    <button class="delete"></button>
+                </div>
+                <div class="numOfLikes">Likes: ${post.likes}</div>
+                <div class="description">
+                    ${post.description}
```
Comment (ala-n) [inline UI/scripts/scriptWithDOM.js]: Unsave interpolation

File: UI/scripts/scriptWithDOM.js (hunk 3 above)
Comment (ala-n) [inline UI/scripts/scriptWithDOM.js]: Better use data attributes instead or add a prefix

File: UI/scripts/scriptWithDOM.js [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+                </div>
+            </div>`;
+
+    }
+
+    newID(n) {
```
Comment (ala-n) [inline UI/scripts/scriptWithDOM.js]: Should be static as it doesn't access class state

File: UI/scripts/scriptWithDOM.js [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+        let tmp = (''+n).slice('');
+        let res = '';
+        for(let i = 0; i < 7 - tmp.length; i++)
```
Comment (ala-n) [inline UI/scripts/scriptWithDOM.js]: Why don't  you use something easier?
No comments about process here.
Better use functional style

File: UI/scripts/scriptWithDOM.js [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+    addPost(post) {
+        if (this._validatePost(post)) {
+            this.numOfPosts++;
+            localStorage.setItem('posts_number', JSON.stringify(this.numOfPosts));
+            post.id = this.newID(this.numOfPosts);
+            this._photoPosts.push(post);
+            document.getElementById('posts').innerHTML = this.createHTMLPost(post) + document.getElementById('posts').innerHTML;
+            localStorage.setItem(post.id, JSON.stringify(post));
+            return true;
+        } else return false;
```
Comment (ala-n) [inline UI/scripts/scriptWithDOM.js]: redundant else state

File: UI/scripts/scriptWithDOM.js [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+    }
+
+    updateLent(){
+        document.getElementById('user_name').innerHTML = this.currentUser;
+        document.getElementById('posts').innerHTML = '';
+        let post;
+        for (let i = 1; i <= this.numOfPosts ; i++) {
```
Comment (ala-n) [inline UI/scripts/scriptWithDOM.js]: Unoptimized rendering

File: UI/scripts/scriptWithDOM.js [hunk 9]
```diff
@@ -0,0 +1,323 @@
+class postsView {
```
[the remark this answers is not in the document]
Comment (GlebKast) [reply UI/scripts/scriptWithDOM.js]: I've done DOM before deadline. I've done pull request last tuesday but the task was added to new branch on sunday: 
image

[the remark this answers is not in the document]
Comment (lexastef) [reply UI/scripts/scriptWithDOM.js]: There were no any comments regarding dedlines :) only code issues.

**Pull request merged**